client/app/cloud/project/compute/infrastructure/privateNetwork/cloud-project-compute-infrastructure-privateNetwork-list.directive.js

```diff
@@ -76,6 +89,46 @@ class PrivateNetworkListCtrl {
             .finally(() => this.loaders.vrack.get = false);
     }
 
+
+    /**
+     * open UI activate private network modal if there are pre ordered vRacks else
+     * take user to order new vRack page
+     *
+     * @memberof PrivateNetworkListCtrl
+     */
+    addVRack () {
+        var vRacks = this.vRacks;
+        if(vRacks && _.isArray(vRacks) && !_.isEmpty(vRacks)) {
```

Comment:
vracks.length > 0

We don't expect vracks to be anything else than an array.  If, for some reasons, it is not an array.. just let the javascript break.  (That way dev/user won't wonder why they get redirected to vrack order even though they have vracks)

Reply:
done